authentication-profile/src/main/java/com/esri/arcgisruntime/sample/authenticationprofile/MainActivity.java

```diff
@@ -0,0 +1,131 @@
+/* Copyright 2016 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.authenticationprofile;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Locale;
+import java.util.concurrent.ExecutionException;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalInfo;
+import com.esri.arcgisruntime.portal.PortalUser;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+
+public class MainActivity extends AppCompatActivity {
+
+    private TextView userText;
+    private TextView emailText;
+    private TextView urlText;
+    private TextView createDate;
+    private ImageView userImage;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // Set the DefaultAuthenticationChallegeHandler to allow authentication with the portal.
+        DefaultAuthenticationChallengeHandler handler = new DefaultAuthenticationChallengeHandler(this);
+        AuthenticationManager.setAuthenticationChallengeHandler(handler);
+        // Set loginRequired to true always prompt for credential,
+        // When set to false to only login if required by the portal
+        final Portal portal = new Portal("http://www.arcgis.com", true);
+        portal.addDoneLoadingListener(new Runnable() {
+            @Override
+            public void run() {
+                if (portal.getLoadStatus() == LoadStatus.LOADED) {
+                    // Get the portal information
+                    PortalInfo portalInformation = portal.getPortalInfo();
+                    String portalName = portalInformation.getPortalName();
+                    urlText = (TextView)findViewById(R.id.portal);
+                    urlText.setText(portalName);
+
+                    // Get the authenticated portal user
+                    PortalUser user = portal.getUser();
```

Comment:
Should we check if `user` is not null? if our users modify this sample and change `loginRequired` to false, the portal may be loaded without login. Then `user` will be null. Therefore NPE in the following. Maybe it is ok to leave it as it is. your call.